### PR TITLE
ocamlPackages.ppx_deriving_yaml: init at 0.1.0

### PR DESCRIPTION
**Pull request by tg-x**
###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/ocaml-modules/ppx_deriving_yaml/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+, ppx_deriving, yaml
+}:
+
+buildDunePackage rec {
+  pname = "ppx_deriving_yaml";
+  version = "0.1.0";
+
+  useDune2 = true;
+
+  minimumOCamlVersion = "4.07";
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/ppx_deriving_yaml/default.nix:11]:
```suggestion
  minimalOCamlVersion = "4.08";
```

File: pkgs/development/ocaml-modules/ppx_deriving_yaml/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+  src = fetchFromGitHub {
+    owner = "patricoferris";
+    repo = "ppx_deriving_yaml";
+    rev = "v${version}";
+    sha256 = "1pdxvkm7hd4iikfzndacj2xymmwz8py07lrnyyn7jm1x0619j1qf";
+  };
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/ppx_deriving_yaml/default.nix:18]: You can use `fetchurl` and download the source from there: https://github.com/patricoferris/ppx_deriving_yaml/releases/download/v0.1.0/ppx_deriving_yaml-v0.1.0.tbz

The released artifact prepared by the authors in particular includes correct version information.

Comment (tg-x) [reply pkgs/development/ocaml-modules/ppx_deriving_yaml/default.nix:18]: oh ok i thought fetchfromgithub was the preferred method

**Comment by tg-x** [comment]
@vbgl I have applied the suggested changes

File: pkgs/development/ocaml-modules/ppx_deriving_yaml/default.nix (hunk 1 above)
Comment (vbgl) [reply pkgs/development/ocaml-modules/ppx_deriving_yaml/default.nix]: The suggestion to fix the spelling still holds.

File: pkgs/development/ocaml-modules/ppx_deriving_yaml/default.nix (hunk 2 above)
Comment (vbgl) [reply pkgs/development/ocaml-modules/ppx_deriving_yaml/default.nix:16]: As `fetchurl` does not have a `meta.homepage` attribute, you must define it for your derivation.

**Pull request merged**